Foundation/URLCache.swift

```diff
@@ -142,13 +147,37 @@ open class URLCache : NSObject {
     */
     open class var shared: URLCache {
         get {
-            NSUnimplemented()
+            return sharedSyncQ.sync {
+                if let cache = sharedCache {
+                    return cache
+                } else {
+                    let fourMegaByte = 4 * 1024 * 1024
+                    let twentyMegaByte = 20 * 1024 * 1024
+                    let bundleIdentifier = Bundle.main.bundleIdentifier ?? UUID().uuidString
```

Comment:
That `UUID` at the end is a little bit worrying. The app will fill caches, and will never reuse the cached results, filling the disk of garbage. I can think at least of `ProcessInfo.processInfo.processName` as a good fallback, but it is not unique (but the bundle identifier doesn’t have to be either). Maybe if no bundle identifier is found the best thing is not to have a disk cache at all.

Reply:
Can this point to the temporary directory with a fixed name? (A sanitized process name does sound good to me!)